File: test/react-native/features/fixtures/scenario-launcher/android/src/main/java/com/bugsnag/reactnative/test/scenarios/ContextNativeCustomScenario.kt
package com.bugsnag.reactnative.test

import android.content.Context
import com.bugsnag.android.Bugsnag
import com.facebook.react.bridge.Promise

class ContextNativeCustomScenario(context: Context): Scenario(context) {

    override fun run(promise: Promise) {
        Bugsnag.notify(generateException())
        Thread.sleep(500)
        Bugsnag.setContext("context-native")
        Bugsnag.notify(RuntimeException(javaClass.simpleName + "2"))
    }
}

File: test/react-native/features/fixtures/scenario-launcher/src/scenarios/ContextNativeCustomScenario.js
import Scenario from './Scenario'
import Bugsnag from '@bugsnag/react-native'
import { NativeInterface } from '../lib/native'

export class ContextNativeCustomScenario extends Scenario {
  run () {
    Bugsnag.setContext('context-js')
    setTimeout(() => {
      NativeInterface.runScenario('ContextNativeCustomScenario')
    }, 100)
  }
}
